### PR TITLE
Parse Slack usernames

### LINKED ISSUE #52: Parse Slack usernames in EODs
User mentions (like this `@belavina`) are displayed as `<@UMYID|belavina>` on the dashboard.

We should either strip away `<@UMYID|>` part or use Slack API to get user `real_name`: https://api.slack.com/methods/users.list

### PR DESCRIPTION
**Pull request by scrust**
Parse Slack usernames based on Slack's directives for [retrieving formatted messages](https://api.slack.com/messaging/composing/formatting#retrieving-messages).

The string `<@UJFPKA3A4|stuart.crust>` now appears as **stuart.crust** in the dashboard. I'm open to suggestions on formatting, but I figured bold would be a good start to differentiate the names. The name after the `|` in the mention matches the `username` displayed, so it didn't require a separate API call to get the data.

Also did some cleanup on the `Osteppy` component.

Closes #52.

### CODE REVIEW

File: Dashboard/src/components/Osteppy.js [hunk 1]
```diff
@@ -3,24 +3,66 @@ import Panel from './common/Panel';
...
+import Moment from 'moment';
+
 const COMPONENT_NAME = "osteppy";
 
+// Reformat Slack messages based on:
+// https://api.slack.com/messaging/composing/formatting#retrieving-messages
+function reformatSlackMentions(text) {
+  return text.replace(/<(.*?)>/g, (match, p1) => {
+    switch(p1[0]) {
+      case '@': // content starting with `@U` or `@W` is a user mention
```
Comment (belavina) [inline Dashboard/src/components/Osteppy.js]: @scrust Do you mind applying it to the channels as well? `#channel` turns into `<#C64TOPQHJ|channel` (sorry didn't mention in the issue).

One issue with just stripping the tags away is that sometimes the `ID` returned is not equal to `@username`. I think this should work for now though

File: Dashboard/src/components/Osteppy.js [hunk 2]
```diff
@@ -3,24 +3,66 @@ import Panel from './common/Panel';
...
+        console.warn(`Unknown mention: ${match}`);
+        return match
+    }
+  });
+}
+
+function EODList(props) {
+ return (
+   <>
+    <h3>{props.title}</h3>
```
Comment (belavina) [inline Dashboard/src/components/Osteppy.js]: Sometimes PastEODS/PresentEODS can be empty (so in this case there's will be EOD headers followed by the empty content).

File: Dashboard/src/components/Osteppy.js (hunk 1 above)
Comment (scrust) [reply Dashboard/src/components/Osteppy.js]: Can definitely add channel parsing, no problem!

The `ID` isn't usually equal to the `@username`, is it? E.g. I'm `@scrust` but my `ID` is `stuart.crust`; `stuart.crust` is what appears in red for me:

image

I figured making the names consistent, at least on the dashboard, would be the least confusing.

File: Dashboard/src/components/Osteppy.js (hunk 2 above)
Comment (scrust) [reply Dashboard/src/components/Osteppy.js]: Ah, so the headers shouldn't display if there's no EODs for that group? Gotcha 👍

File: Dashboard/src/components/Osteppy.js (hunk 1 above)
Comment (belavina) [reply Dashboard/src/components/Osteppy.js]: For some users slack id is quite different from what appears with `@username`.
I think we will keep the `ID`s for consistency for now.

**Comment by scrust** [comment]
Here's what it looks like now:

image

I prepended a `#` to the channel name so it's clearer it's referring to a channel, and I'm still very open to changing the formatting if we want to add colours, etc.

**Comment by belavina** [comment]
@scrust Looks good enough to me. Feel free to add colors if you wish or links to the channels themselves (https://api.slack.com/docs/deep-linking#app_channel)

**Pull request merged**